### PR TITLE
conftest 0.23.0 (new formula)

### PR DESCRIPTION
**Pull request by jpreese**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Conftest // https://github.com/open-policy-agent/conftest

### CODE REVIEW

File: Formula/conftest.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  homepage "https://www.conftest.dev/"
+  url "https://github.com/open-policy-agent/conftest/archive/v0.23.0.tar.gz"
+  sha256 "116bfd3c00fcaa31a3468cace6df5b03eeff0150d0d096aee53dd74c7c9647bd"
+  license "Apache-2.0"
+  head "https://github.com/open-policy-agent/conftest.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-ldflags", "-X github.com/open-policy-agent/conftest/internal/commands.version=#{version}"
```
Comment (carlocab) [inline Formula/conftest.rb:12]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-X github.com/open-policy-agent/conftest/internal/commands.version=#{version}")
```

File: Formula/conftest.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    assert_match "Test your configuration files using Open Policy Agent", shell_output("#{bin}/conftest --help")
```
Comment (carlocab) [inline Formula/conftest.rb:16]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @jpreese.

File: Formula/conftest.rb (hunk 2 above)
Comment (jpreese) [reply Formula/conftest.rb:16]: test for the `verify` command has been added

**Comment by jpreese** [comment]
Thanks for the review, @carlocab ! A test has been added for the verify command.

File: Formula/conftest.rb (hunk 1 above)
Comment (carlocab) [reply Formula/conftest.rb:12]: Looks like you got rid of the commit that fixed this by accident. When you fix this, please squash all your commits together with the first one.

Comment (jpreese) [reply Formula/conftest.rb]: Re-added and squashified.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**